addons/controls/README.md

```diff
@@ -0,0 +1,230 @@
+
+export const Basic = () => <Button label="hello" />;
+```
+
+You could simply rewrite it as:
+
+```js
+export const Basic = (args = { label: 'hello' }) => <Button {...args} />;
+```
+
+### Manually-configured args
+
+Auto-generated args make sense when the controls correspond exactly to component. But there are cases where you want controls to control something about the story instead of (or in addition to) the component properties.
+
+Consider the following example that shows reflow behavior of an array of `Button` instances inside a `ButtonGroup` container:
+
+```js
+import range from 'lodash/range';
+import { Button, ButtonGroup } from '.';
+
+export const Reflow = ({ count }) => (
+  <ButtonGroup>
+    {range(count).map((i) => (
+      <Button label={`button ${i}`} />
+    ))}
+  </ButtonGroup>
+);
+```
+
+In this case, we need to specify `ArgTypes` metadata about the story to render properly:
+
+```js
+Reflow.story = {
+  argTypes: {
+    count: { control: { type: 'range', min: 0, max: 100 } },
+  },
+};
+```
+
+This is the same kind of data that gets filled in automatically in the [auto-generated args case](#auto-generated-args).
+
+## Migrating from knobs
+
+If you're already using [Storybook Knobs](https://github.com/storybookjs/storybook/tree/master/addons/knobs) you should consider migrating to Controls.
+
+You're probably using it for something that can be satisfied by one of the two previous cases, or [manually-configured args](#manually-configured-args).
+
+Let's consider a few different knobs-based stories.
+
+### Knobs to auto-generated args
+
+First, let's consider a knobs version of a basic story that fills in the props for a component:
+
+```js
+import { text } from '@storybook/addon-knobs';
+import { Button } from './Button';
+
+export const Basic = () => <Button label={text('Label', 'hello')} />;
+```
+
+This fills in the Button's label based on a knob, which is exactly the [auto-generated](#auto-generated-args) use case above. So we can rewrite it using auto-generated args:
+
+```js
+export const Basic = (args = { label: 'hello' }) => <Button {...args} />;
+```
+
+### Knobs to manually-configured args
+
+Similarly, we can also consider a story that uses knob inputs to change its behavior:
+
+```js
+import range from 'lodash/range';
+import { number } from '@storybook/addon-knobs';
+
+export const Reflow = () => {
+  const count = number('Count', 10, { min: 0, max: 100, range: true });
+  return (
+    <ButtonGroup>
+      {range(count).map((i) => (
+        <Button label={`button ${i}`} />
+      ))}
+    </ButtonGroup>
+  );
+};
+```
+
+And again, as above, this can be rewritten using [manually-configured args](#manually-configured-args):
+
+```js
+export const Reflow = ({ count }) => (
+  <ButtonGroup>
+    {range(count).map((i) => (
+      <Button label={`button ${i}`} />
+    ))}
+  </ButtonGroup>
+);
+Reflow.story = {
+  argTypes: {
+    count: { control: { type: 'range', min: 0, max: 100 } },
```

Comment:
I think there is a mixed message in this example, because not only you're adding the argTypes, but also there seems to be a new mechanic within the control API that translates perfectly the use case of `lodash/range`. Users who don't fit in this use case might either be confused about this change or get it and wonder what other types are available